tdrs-backend/tdpservice/users/views.py

```diff
@@ -26,14 +27,15 @@ class UserViewSet(
 ):
     """User accounts viewset."""
 
-    queryset = User.objects.select_related("stt")
+    queryset = User.objects\
+        .select_related("stt")\
+        .prefetch_related("groups__permissions")
 
     def get_permissions(self):
         """Get permissions for the viewset."""
-        if self.action == "create":
-            permission_classes = [AllowAny]
-        else:
-            permission_classes = [IsUserOrReadOnly]
+        permission_classes = {"create": [AllowAny], "list": [IsAdmin]}.get(
```

Comment:
I read this as saying that anyone can create a user object via the API without authentication, which doesn't seem like what we'd want. 

Since we have tests showing that unauthorized users can't create roles or permissions via the API, could we add a similar test for users to lock this down?

Reply:
@alexsoble Actually, this is how we are able to create an account for a user who doesn't yet exist in the system. Without this, the first time a user logged in to the system, it would block the system from creating a profile. Since one of our requirements was to automatically create a profile when the user logs in the first time, this is necessary to allow it.

Reply:
Hmmmm, when I put `"create": [AllowAny]` on its own line locally and then comment it out and run the test suite locally, I don't see any additional test failures — do you see the same thing?

Reply:
@alexsoble Good catch on this one! This is now updated to what it should be, and will error on tests if you take out the "AllowAny" - The way it was set up it turned out a DRF base permission class was returning that it had permission instead of it checking our class. It's all working as expected now.

Reply:
Gotcha! When I look at `login.py`, here's the method I see that's creating new users:

```
    def handle_user(self, request, id_token, decoded_payload):
        """Handle the incoming user."""
        # get user from database if they exist. if not, create a new one
        if "token" not in request.session:
            request.session["token"] = id_token

        user = CustomAuthentication.authenticate(
            self, username=decoded_payload["email"]
        )
        if user is not None:
            self.login_user(request, user, "User Found")
        else:
            User = get_user_model()
            user = User.objects.create_user(decoded_payload["email"])
            user.set_unusable_password()
            user.save()
            self.login_user(request, user, "User Created")
```

It doesn't look like this method is hitting the API endpoint. 

So if we remove the API endpoint, would the create-new-user functionality break or stay the same? If I'm not looking at the right file or method please point me to the right place in the code.

Reply:
Zooming out one level, I have a question about the longer mapping of actions to permissions classes in 1c5b4f8f6ad7851732ffa84fa6fc57b3dd7598c7. 

Do we want to be hard-coding what permissions belong to what groups? 

My understanding is that would be set or editable via the Django Admin interface, e.g.: 

![screenshot of Django Admin dashboard](https://raw.githubusercontent.com/raft-tech/TANF-app/backend/309/dynamic-roles-permissions/tdrs-backend/docs/images/group_permissions.png)

Reply:
Hmmmm, this is starting to clear up for me a bit. 

There are two different sources of permissions being used in this app: permissions built in to Django, and permissions built into Django Rest Framework.

My understanding is that the Django permissions are stored in the database and can be edited via the Django Admin Interface, while the Django Rest Framework permissions aren't stored in the DB and aren't editable right now, is that correct?

Reply:
That is one way of looking at it - Django Model Permissions are loaded to the database automatically and manage CRUD permissions for all models, while DRF has some permissions you can use in the code such as `AllowAny` and `IsAuthenticated`.

I think that our long term strategy will be to implement default permissions leveraging the Django Model Permissions as it will give us a more granular approach to permissions, but implementing that is outside the scope of this issue.

Reply:
> So if we remove the create user API endpoint, would the create-new-user-on-login functionality break or stay the same? If I'm not looking at the right file or method please point me to the right place in the code.

I believe the "create-new-user-on-login" functionality would most likely remain the same, but I'd want to test it to be sure, if it's a priority.

Reply:
> Do we want to be hard-coding what permissions belong to what groups?
> 
> My understanding is that would be set or editable via the Django Admin interface, e.g.:

No, we don't want to hard code what permissions belong to what groups. Using the Django Admin is definitely a better approach for assigning permissions to groups.

Reply:
Just noting that we documented removing the "create" endpoint in a separate issue here: 

https://app.zenhub.com/workspaces/tdrs-product-backlog-5f2c6cdc7c0bb1001bdc43a5/issues/raft-tech/tanf-app/518